tilequeue/process.py

```diff
@@ -443,3 +443,104 @@ def process_coord(coord, nominal_zoom, feature_layers, post_process_data,
         unpadded_bounds, cut_coords, buffer_cfg, extra_data, scale)
 
     return all_formatted_tiles, extra_data
+
+
+def convert_source_data_to_feature_layers(rows, layer_data, bounds, zoom):
+    # TODO we might want to fold in the other processing into this
+    # step at some point. This will prevent us from having to iterate
+    # through all the features again.
+
+    layers = (
+        'boundaries',
+        'buildings',
+        'earth',
+        'landuse',
+        'places',
+        'pois',
+        'roads',
+        'transit',
+        'water',
+    )
+
+    features_by_layer = {}
+    for layer in layers:
+        features_by_layer[layer] = []
+
+    for row in rows:
+
+        fid = row.pop('__id__')
+
+        geometry = row.pop('__geometry__', None)
+        label_geometry = row.pop('__label__', None)
+        boundaries_geometry = row.pop('__boundaries_geometry__', None)
+        assert geometry or boundaries_geometry
+
+        common_props = row.pop('__properties__', None) or {}
```

Comment:
I just confused myself too, so I expanded it out and added a comment in da19aa0.